### PR TITLE
Avoid using `__align__` in ROCm

### PR DESCRIPTION
**Pull request by kmaehashi**
Ref #2562

### CODE REVIEW

**Comment by leofang** [comment]
Geez...I wonder if the template `complex_storage` in the latest Thrust would work with ROCm? If so, we get another incentive for resolving #2629 and then revisit the alignment issue...

File: cupy/core/include/cupy/complex/complex.h [hunk 1]
```diff
@@ -62,7 +62,12 @@ struct _select_greater_type
  *
  */
 template <typename T>
+#if defined(__CUDACC__)
```
Comment (leofang) [inline cupy/core/include/cupy/complex/complex.h]: Can you also test `__CUDACC_RDC__` here? It's a preparation for #2426.

**Comment by Crissman** [comment]
imosCI, test this please.

**Comment by Crissman** [comment]
imosCI, test this please.

File: cupy/core/include/cupy/complex/complex.h (hunk 1 above)
Comment (kmaehashi) [reply cupy/core/include/cupy/complex/complex.h]: I think both `__CUDACC__` and `__CUDACC_RDC__` are set when built with `--relocatable-device-code=true`.

Comment (leofang) [reply cupy/core/include/cupy/complex/complex.h]: Ah, I see! Yes, I can verify. Thanks @kmaehashi! This is not obvious from the documentation...☹️

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**